routes/songRoutes.js

```diff
@@ -0,0 +1,156 @@
+        return res
+          .status(404)
+          .json({ message: "Song not found, please try another id" });
+      }
+
+      // Send the retrieved song as a JSON response
+      res.json(song);
+    })
+    .catch((error) => {
+      // If an error occurs during the database query, send an error response
+      res
+        .status(500)
+        .json({ message: "Internal server error", error: error.message });
+    });
+});
+
+// ------- One Metallica song based on title -------
+// Show a specific Metallica song based on the song title (case-insensitive and partial match)
+router.get("/song-name/:title", (req, res) => {
+  const songTitle = req.params.title.toLowerCase(); // Convert to lowercase for case-insensitive search
+
+  // Find a song by its title using the Mongoose model, allowing partial matches
+  MetallicaSongModel.find({ name: { $regex: songTitle, $options: "i" } })
+    .then((songs) => {
+      // If no song titles are found, send a 404 error response
+      if (!songs || songs.length === 0) {
+        return res
+          .status(404)
+          .json({ message: "Song not found, please try another song title" });
+      }
+
+      // Send the retrieved song title(s) as a JSON response
+      res.json(songs);
+    })
+    .catch((error) => {
+      // If an error occurs during the database query, send an error response
+      res
+        .status(500)
+        .json({ message: "Internal server error", error: error.message });
+    });
+});
+
+// ------- One Metallica album -------
+// Show a specific Metallica album based on the album title (case-insensitive and partial match). Also has the possibillity to check the popularity of the songs within an album with query param
+router.get("/albums/:album", (req, res) => {
+  const requestedAlbum = req.params.album.toLowerCase(); // Convert to lowercase for case-insensitive search
+  const popularity = parseInt(req.query.popularity, 10); // Parse the 'popularity' query parameter as an integer
+
+  // Find the album by title in the database
+  MetallicaSongModel.find({ album: { $regex: requestedAlbum, $options: "i" } })
+    .then((foundAlbum) => {
+      // If no album found, send a 404 error response
+      if (!foundAlbum || foundAlbum.length === 0) {
+        return res
+          .status(404)
+          .json({ message: "Album not found, please try another album title" });
+      }
+
+      // If popularity query parameter exists and is valid, filter songs by popularity
+      if (!isNaN(popularity) && popularity >= 0) {
+        // Filter songs within the album by popularity
+        const songsWithDesiredPopularity = foundAlbum.filter(
+          (song) => song.popularity >= popularity
+        );
+
+        // If no songs match the popularity criterion, send an error response
+        if (songsWithDesiredPopularity.length === 0) {
+          return res.status(404).json({
+            message:
+              "No songs found with this popularity or more in this album",
```

Comment:
...and this error handling are well set up and guides the user in a clear way on what has gone wrong when searching for an album and songs with a certain popularity within it.